subworkflows/local/bam_merge_index_samtools/main.nf

```diff
@@ -7,6 +7,34 @@
 include { SAMTOOLS_INDEX as INDEX_MERGE_BAM } from '../../../modules/nf-core/samtools/index/main'
 include { SAMTOOLS_MERGE as MERGE_BAM       } from '../../../modules/nf-core/samtools/merge/main'
 
+
+process SORT_BAM {
+    
+    input:
+
+    tuple val(meta), path(input_files)
+ 
+    output:
+
+    tuple val(meta), path(output_files)
+
+    script:
+
+    // BAM files are staged in subdirs of the work directory:
+    //   1/file_234.bam 
+    //   2/file_123.bam
+    //   ...
+    // We want to sort based on the filename, so we remove
+    // the relative path to the bam file for the sort criterion.
+
+    output_files = input_files.sort {it.name.replaceAll('.*/','')}
```

Comment:
Is it possible to put the sort logic [here](https://github.com/eitm-org/sarek/blob/master/workflows/sarek.nf#LL414C18-L414C18) instead of creating a new process? Like `.groupTuple(sort: {it.name.replaceAll('.*/',''))`? It might be possible based on the [documentation](https://www.nextflow.io/docs/latest/operator.html#collectfile) but I'm not sure,

Reply:
Yes, great idea.  The groupTuple operator doesn't work, but the map operator is what we want.  I've made the changes in sarek.nf and reverted all changes in subworkflows/local/bam_merge_index_samtools/main.nf.